server/resp/src/main/java/org/infinispan/server/resp/commands/list/LMOVE.java

```diff
@@ -0,0 +1,197 @@
+import org.infinispan.server.resp.commands.Resp3Command;
+import org.infinispan.server.resp.logging.Log;
+import org.infinispan.util.concurrent.CompletionStages;
+import org.infinispan.util.logging.LogFactory;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * @link https://redis.io/commands/lmove/
+ * <p>
+ * Atomically returns and removes the first/last element (head/tail depending on the wherefrom argument)
+ * of the list stored at source, and pushes the element at the first/last element (head/tail depending on the whereto argument)
+ * of the list stored at destination.
+ * <p>
+ * For example: consider source holding the list a,b,c, and destination holding the list x,y,z.
+ * Executing LMOVE source destination RIGHT LEFT results in source holding a,b and destination holding c,x,y,z.
+ * <p>
+ * If source does not exist, the value nil is returned and no operation is performed.
+ * If source and destination are the same, the operation is equivalent to removing the first/last element
+ * from the list and pushing it as first/last element of the list,
+ * so it can be considered as a list rotation command (or a no-op if wherefrom is the same
+ * as whereto).
+ * <p>
+ * This command comes in place of the now deprecated RPOPLPUSH.
+ * Doing LMOVE RIGHT LEFT is equivalent.
+ * @since 15.0
+ */
+public class LMOVE extends RespCommand implements Resp3Command {
+   private final static Log log = LogFactory.getLog(LMOVE.class, Log.class);
+   public static final String LEFT = "LEFT";
+   public static final String RIGHT = "RIGHT";
+
+   public LMOVE() {
+      super(5, 1, 2, 1);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler, ChannelHandlerContext ctx,
+                                                                List<byte[]> arguments) {
+
+      if (arguments.size() != 4) {
+         // ERROR
+         RespErrorUtil.wrongArgumentNumber(this, handler.allocatorToUse());
+         return handler.myStage();
+      }
+
+      return lmoveAndReturn(handler, ctx, arguments, false);
+   }
+
+   protected CompletionStage<RespRequestHandler> lmoveAndReturn(Resp3Handler handler,
+                                                                ChannelHandlerContext ctx,
+                                                                List<byte[]> arguments,
+                                                                boolean rightLeft) {
+      byte[] source = arguments.get(0);
+      byte[] destination = arguments.get(1);
+
+      boolean txCache = handler.cache().getCacheConfiguration().transaction().transactionMode().isTransactional();
+      boolean sameList = source.equals(destination);
+
+      if (!sameList && !txCache) {
+         // different lists and not tx cache, warn
+         log.warn("LMOVE command can't guarantee atomicity and consistency when the source list "
```

Comment:
log once

Reply:
https://github.com/infinispan/infinispan/blob/main/core/src/main/java/org/infinispan/util/logging/Log.java#L1934

Reply:
We also may want to log once per cache, which would preclude the use of the jboss logging annotation sadly.

Reply:
how should I do this @wburns ?